.github/workflows/integration.yaml

```diff
@@ -0,0 +1,46 @@
+name: Integration Tests
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: "1.21"
+      - name: tests
+        run: |
+          echo "stop/disable/kill mono"
+          sudo systemctl stop mono-xsp4.service || true
+          sudo systemctl disable mono-xsp4.service || true
+          sudo pkill mono || true
+          echo "change to weather example directory"
+          cd example/weather
+          echo "run setup script"
+          ./scripts/setup
+          echo "run server"
+          ./bin/forecaster -monitoring-enabled=false &
+          ./bin/locator -monitoring-enabled=false &
+          ./bin/tester -monitoring-enabled=false &
+          ./bin/front -monitoring-enabled=false &
+          sleep 1
```

Comment:
I wonder if we couldn't instead wait for the service health checks to start responding? Using `sleep` really isn't great... Maybe we could introduct a function such as:
```bash
check_service_health() {
    local health_url="$1"
    local start_time=$(date +%s)

    while : ; do
        if curl -s --fail "$health_url" > /dev/null; then
            echo "Service is up!"
            return 0
        fi

        local current_time=$(date +%s)
        if (( current_time - start_time >= 5 )); then
            echo "Timed out waiting for service to be up."
            return 1
        fi

        sleep 0.2
    done
}
```
and then call it for each service concurrently and wait for the calls to return:
```bash
check_service_health "http://localhost:????/healthz" &
check_service_health "http://localhost:????/healthz" &
check_service_health "http://localhost:????/healthz" &
check_service_health "http://localhost:????/healthz" &
wait
```